src/main/java/duke/command/FindTaskCommand.java

```diff
@@ -0,0 +1,28 @@
+package duke.command;
+
+import duke.DukeException;
+import duke.storage.Storage;
+import duke.task.TaskList;
+import duke.ui.TextUi;
+
+public class FindTaskCommand extends Command {
+    private static final boolean IS_EXIT = false;
+    private final String keywords;
+
+    public FindTaskCommand(String keywords) {
+        super(IS_EXIT);
+        this.keywords = keywords;
+    }
+
+    @Override
+    public void execute(TaskList tasks, TextUi ui, Storage storage) {
+        try {
+            TaskList filteredTasks = tasks.filterTask(keywords);
+            for (int i = 0; i < filteredTasks.getSize(); i++) {
+                ui.showToUser((i + 1) + ". " + filteredTasks.getTask(i));
+            }
```

Comment:
Perhaps encapsulate this part of the code into a function to reduce code duplication?

```suggestion
            printFilteredTasks();
```